internal/services/postgres/postgresql_server_resource.go

```diff
@@ -663,21 +697,41 @@ func resourcePostgreSQLServerUpdate(d *pluginsdk.ResourceData, meta interface{})
 	}
 
 	// Update Admin Password in the separate call when Replication is stopped: https://github.com/Azure/azure-rest-api-specs/issues/16898
-	if d.HasChange("administrator_login_password") && !replicaUpdatedToDefault {
-		properties.Properties.AdministratorLoginPassword = utils.String(d.Get("administrator_login_password").(string))
+	if (d.HasChange("administrator_login_password") || d.HasChange("administrator_login_password_version")) && !replicaUpdatedToDefault {
```

Comment:
Was there a specific reason for using `HasChange` instead of using `HasChanges`? 

`administrator_login_password_version` should be `administrator_login_password_wo_version` as well

```suggestion
	if d.HasChanges("administrator_login_password", "administrator_login_password_wo_version") && !replicaUpdatedToDefault {
```

Reply:
Nope! I sometimes just forget about the existence of certain functions